starknet_devnet/blocks.py

```diff
@@ -90,7 +90,7 @@ def get_state_update(self, block_hash=None, block_number=None) -> BlockStateUpda
 
     async def generate(
         self, transaction: DevnetTransaction, state: StarknetState,
-        state_root: bytes, state_update = None
+        state_root: bytes, state_update = None, is_empty_block = False
```

Comment:
If the linter is reporting extra arguments here, it means it exited with a non-zero status code and the CI/CD is going to fail. One possible fix is to put a pylint suppression just for this function, but maybe even better would be to exclude the `state_root` argument and extract it from `state` inside this function, if possible.

Reply:
alrite! on it!

Reply:
i've inspect the `state` but it doesn't seem to contain `state_root`

i think i'll suppress the linter specifically for this function

Reply:
Look at the usage of the generate function and what's passed into it, should be something like `state.state.shared_state.contract_states`

Reply:
oh yeah thats correct! 

thank you!

Reply:
there's even a getter function for that, apology for missing that function

done!

Reply:
Great, I see the getter can be removed now that it's not used. Will approve and merge.